docs/misc/ConfigDB_state.md

```diff
@@ -0,0 +1,89 @@
+* NotConfigured -> BrokenLink       ERROR - throw exception
+
+### BrokenLink handling
+BrokenLink is an error condition. We do not know what was the admin intent so we fail all Docker volume requests until the situation is resolved by the admin using `config rm --local` command
+
+* BrokenLink -> NotConfigured - by admin `config rm` command
+* BrokenLink -> BrokenLink -  noop
+* BrokenLink -> SingleNode or MultiNode - no direct transition, the Config DB has to go through NotConfigured first.
+  * However, a vDVS service may first find out the state is BrokenLink, then admin does 2 commands and then vDVS service may directly transfer it's internal state to MultiNode or SingleNode
+
+
+### From SingleNode
+
+Transition from SingleNode means the admin live-reconfigured the Config DB.
+On the node where the operation happened, the admin CLI will refresh the vDVS service which will find out the new state from the DB.
+
+On other nodes, the discovery will find out about the change to `MultiNode` only, and will issue a warning to the admin  - since we do not know what was the intent.
+
+* SingleNode    -> Single Node      noop
+* SingleNode    -> NotConfigured    happened on `config -rm --local` only. Restart the service.
+* SingeNode     -> MultiNode        can happen on discovery. WARNING - Config changed. No action  we do not know what was the intent.
+* SingeNode     -> BrokenLink       ERROR - Throw exception
+
+### From MultiNode
+
+To transition from MultiNode, either the admin has to issue `config --rm` or there should be a FS or DB issue (i.e. the symlink was accidentally removed)
+
+* MultiNode      -> MultiNode       noop
+* MultiNode      -> NotConfigured   restart the service (on discovery or on CLI)
+* MultiNode      -> SingleNode      same
+* MultiNode      -> BrokenLink      ERROR - Throw exception
+
+## Discovery
+
+Discovery goal is to sync `vmdk_ops.py` Config DB state to reflect changes to the state done on a different ESXi node.
+
+**The key use case** is initial configuration, where all nodes start with NotConfigured. On one of the nodes, admin (manually or via automation) configures a shared Config DB on a datastore visible to all nodes. Then discovery kicks in on all nodes, finds out there is a fresh Config DB on a shared store, and configures the node to use this DB. This way Admin only need to do configuration on one node and the rest pick it up . See Issue https://github.com/vmware/docker-volume-vsphere/issues/1086 for details.
+
+The only state transition supported by Discovery is NotConfigured -> MultiNode.
+*  Note - once defined, the DB becomes a single point of failure, and the only way to remove it is to run local CLI `config rm --local` on each ESX.
+  * To clarify `config rm --local` removes al `local knowledge` about Config DB - be it the symlink or the actual DB.
+* The Config DB will contain a list of ESX nodes allowed to participate in the discovery (FQDN, FQDN regexp, IPv4 IP or regexp).
```

Comment:
(1) No automatic discovery when there is more than 1 shared DB.  
(2) Not yet. it will be done for discovery. Default "*" , it will be pre-defined and *will* have admin CLI support. We will need to discuss it, I am not sure if we need it, maybe just tracking participants is enough